frontend/src/pages/AccessRequestPage.tsx

```diff
@@ -0,0 +1,228 @@
+  );
+  const request = useSelector<RootState, IGenericNetworkAction>(
+    state => (state.network as any)[actionTypes.ADD_REQUEST_ACCESS] as IGenericNetworkAction,
+  );
+  const agencies = _.filter(lookupCodes, (lookupCode: ILookupCode) => {
+    return lookupCode.type === API.AGENCY_CODE_SET_NAME;
+  });
+  const roles = _.filter(lookupCodes, (lookupCode: ILookupCode) => {
+    return lookupCode.type === API.ROLE_CODE_SET_NAME;
+  });
+
+  let checkAgencies: {} | null | undefined;
+  let checkRoles: {} | null | undefined;
+
+  let accessRequest = data?.accessRequest;
+  let initialValues: IAccessRequestForm = {
+    id: accessRequest?.id ?? 0,
+    userId: userInfo?.sub,
+    user: {
+      id: userInfo?.sub,
+      username: userInfo?.username,
+      displayName: userInfo?.name,
+      firstName: userInfo?.firstName,
+      lastName: userInfo?.lastName,
+      email: userInfo?.email,
+      position: accessRequest?.user?.position ?? userInfo?.position ?? '',
+    },
+    agencies: accessRequest?.agencies ?? [],
+    isGranted: accessRequest?.isGranted ?? false,
+    roles: accessRequest?.roles ?? [],
+    note: accessRequest?.note ?? '',
+    agency: accessRequest?.agencies?.find(x => x !== undefined).id,
+    role: accessRequest?.roles?.find(x => x !== undefined).id,
+    rowVersion: accessRequest?.rowVersion,
+  };
+
+  const mapLookupCode = (code: ILookupCode, defaultId: number | string | null): SelectOption => ({
+    label: code.name,
+    value: code.id.toString(),
+    selected: code.id === defaultId,
+  });
+
+  const selectAgencies = agencies.map(c => mapLookupCode(c, initialValues.agency));
+  const selectRoles = roles.map(c => mapLookupCode(c, initialValues.role));
+
+  checkAgencies = (
+    <Select
+      label="Agency"
+      field="agency"
+      required={true}
+      options={selectAgencies}
+      placeholder={initialValues?.agencies?.length > 0 ? undefined : 'Please Select'}
+    />
+  );
+
+  checkRoles = (
+    <Select
+      label="Role"
+      field="role"
+      required={true}
+      options={selectRoles}
+      placeholder={initialValues?.roles?.length > 0 ? undefined : 'Please Select'}
+    />
+  );
+
+  let button = initialValues.id === 0 ? 'Submit' : 'Update';
+
+  return (
+    <div>
+      <div>
+        <h3>Access Request</h3>
+      </div>
+      <hr />
+      <Container fluid={true}>
+        <Row className="justify-content-md-center">
+          <Formik
+            enableReinitialize={true}
+            initialValues={initialValues}
+            validationSchema={AccessRequestSchema}
+            onSubmit={(values, { setSubmitting }) => {
+              dispatch(getSubmitAccessRequestAction(toAccessRequest(values)));
+              setSubmitting(false);
+            }}
+          >
+            {props => (
+              <Form className="userInfo">
+                {(!request?.isFetching && request?.status === 201) || request?.status === 200 ? (
```

Comment:
We need a better way to do this.  My styling is horribly ugly too.

Reply:
Using an alert would be preferable to a <p>
https://react-bootstrap.github.io/components/alerts/

Reply:
could add some utility functions to request to incapsulate the fetching check and the expected statuses:
request.isCompleteWithStatus(200, 201) or similar.

Reply:
done